File: backend/src/main/java/com/devmatch/backend/exception/ServiceException.java
package com.devmatch.backend.exception;

import com.devmatch.backend.global.RsData;

public class ServiceException extends RuntimeException {

  private final String resultCode;
  private final String msg;

  public ServiceException(String resultCode, String msg) {
    super(resultCode + " : " + msg);
    this.resultCode = resultCode;
    this.msg = msg;
  }

  public RsData<Void> getRsData() {
    return new RsData<>(resultCode, msg, null);
  }
}

File: backend/src/main/java/com/devmatch/backend/exception/ServiceException.kt
package com.devmatch.backend.exception

import com.devmatch.backend.global.RsData


class ServiceException(private val resultCode: String, private val msg: String) : RuntimeException(
    "$resultCode : $msg"
) {
    val rsData: RsData<Void>
        get() = RsData<Void>(resultCode, msg)
}
